gazelle/modules_mapping/generator.py

```diff
@@ -25,16 +25,22 @@ class Generator:
     stderr = None
     output_file = None
     excluded_patterns = None
-    mapping = {}
 
-    def __init__(self, stderr, output_file, excluded_patterns):
+    def __init__(self, stderr, output_file, excluded_patterns, include_stub_packages):
         self.stderr = stderr
         self.output_file = output_file
         self.excluded_patterns = [re.compile(pattern) for pattern in excluded_patterns]
+        self.include_stub_packages = include_stub_packages
+        self.mapping = {}
 
     # dig_wheel analyses the wheel .whl file determining the modules it provides
     # by looking at the directory structure.
     def dig_wheel(self, whl):
+        # Skip stubs and types wheels.
+        wheel_name = get_wheel_name(whl)
+        if wheel_name.endswith(("_stubs", "_types")) and self.include_stub_packages:
```

Comment:
Some type packages _start_ with `types_`, such as https://pypi.org/project/types-WTForms/ and https://pypi.org/project/types-requests/